File: examples/golang/data/initial/ios.go
// +build test

package initial

import (
	"forcedata/examples/golang/model"
)

// データの識別子
const (
	Test01Ios1 model.DataID = "Test01Ios1"
)

func RegisterIos() {
	f := model.Factory
	f.Register(Test01Ios1, func() interface{} {
		return &model.Ios{
			Manufacturer: "apple",
			Model:        "iphone12",
		}
	})
}

File: examples/typescript/data/initial/ios.ts
import { factory } from '../../model/factory';
import { Ios } from '../../model/model';
import { DATAID } from '../dataid';

export function registerInitialIos() {
  const f = factory;
  f.register(DATAID.TEST01_IOS_1, () => {
    return {
      manufacturer: 'apple',
      model: 'iphone12',
    } as Ios;
  });
}
